tap_shiphero/sync.py

```diff
@@ -61,7 +49,19 @@ def sync_products(client, catalog, state, start_date, end_date, stream_id, strea
 
     write_schema(catalog, stream_id)
 
-    last_date = get_bookmark(state, stream_id, start_date)
+    last_date = bookmarks.get_bookmark(state, stream_id, 'datetime', start_date)
+
+    # Rip this out once all bookmarks are converted
+    old_style_bookmark = bookmarks.get_bookmark(state, stream_id, stream_id)
```

Comment:
This doesn't quite match the original implementation. The old style bookmark looks like it was `{'bookmarks':{'stream_id':'2018-04...'}}`, whereas this will check `{'bookmarks':{'stream_id': {'stream_id':'2018-04...'}}}` if I'm reading it correctly.

Reply:
This could actually be `old_bookmark = state['bookmarks']['stream_id'].pop()`. To get and remove simultaneously.